### DISCUSSION TITLE
Emit failed: Error: operation has timed out

### DISCUSSION THREAD

**Discussion by indersein-agilsoft** [socketio/socket.io#4762] [Q&A]
![image](https://github.com/socketio/socket.io/assets/131159015/3b7f0b64-77e3-4e2f-959e-ce4f8741a165)

Please help me how to solve this issue

**Comment by darrachequesne** [comment]
Hi! It seems you are broadcasting with a timeout. A timeout might happen if some clients do not respond to the event within the given delay:

```js
io.timeout(5000).emit("hello", "world", (err, responses) => {
  if (err) {
    // some clients did not acknowledge the event in the given delay
  } else {
    console.log(responses); // one response per client
  }
});
```

Promise-based version:

```js
try {
  const response = await io.timeout(5000).emitWithAck("hello", "world");

  console.log(responses); // one response per client
} catch (e) {
  // some clients did not acknowledge the event in the given delay
}
```

Reference: https://socket.io/docs/v4/broadcasting-events/
